crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -689,24 +698,12 @@ pub fn set_capability_offsets(
     // alphabetic order.
     let mut offset = MAX_RESERVED_MESSAGE_ID + 1;
     for name in shared_capability_names {
-        let version = shared_capabilities.get(&name).unwrap();
-
-        let shared_capability = SharedCapability::new(&name, *version as u8, offset)?;
-
-        match shared_capability {
-            SharedCapability::UnknownCapability { .. } => {
-                // Capabilities which are not shared are ignored
-                debug!("unknown capability: name={:?}, version={}", name, version,);
+        let proto_version = shared_capabilities.get(&name).expect("shared; qed");
 
-                // TODO(mattsse): track shared caps
-            }
-            SharedCapability::Eth { .. } => {
-                // increment the offset if the capability is known
-                offset += shared_capability.num_messages()?;
+        let shared_capability = SharedCapability::new(&name, proto_version.version as u8, offset)?;
 
-                shared_with_offsets.push(shared_capability);
-            }
-        }
+        offset += proto_version.messages;
+        shared_with_offsets.push(shared_capability);
```

Comment:
ah, this sense compared to the match, `SharedCapability` would have had to be extended for people adding new subprotocols. Whereas this is more flexible